### PR TITLE
Remove unused `LogicalPlan::CrossJoin` as it is unused

### LINKED ISSUE #13057: Deprecate `LogicalPlan::CrossJoin`
### Is your feature request related to a problem or challenge?

After https://github.com/apache/datafusion/pull/12985 we don't generate `LogicalPlan::CrossJoin` in the planner.
It would be good to deprecate / remove this in the future.

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by buraksenn**
## Which issue does this PR close?
Closes #13057

## Rationale for this change
LogicalPlan::CrossJoin is not generated anymore so deleted all of its occurences

## What changes are included in this PR?
- delete occurences of crossjoin
- simplify some match statements

## Are these changes tested?
Existing tests

## Are there any user-facing changes?
no

### CODE REVIEW

File: datafusion/expr/src/logical_plan/plan.rs [hunk 1]
```diff
@@ -220,10 +220,6 @@ pub enum LogicalPlan {
     /// Join two logical plans on one or more join columns.
     /// This is used to implement SQL `JOIN`
     Join(Join),
-    /// Apply Cross Join to two logical plans.
-    /// This is used to implement SQL `CROSS JOIN`
-    /// Deprecated: use [LogicalPlan::Join] instead with empty `on` / no filter
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/plan.rs:225]: nice for cleaning up a deprecated feature

**Review by alamb — APPROVED** [review]
THank you @buraksenn  -- this looks like a very nice cleanup to me


cc @Dandandan  who has been working in this area recently as well

**Review by berkaysynnada — APPROVED** [review]
LGTM

**Review by Dandandan — APPROVED** [review]
Let's remove it 🚀

**Pull request merged**